tests/test_app.rs

```diff
@@ -1085,6 +1086,123 @@ fn test_list_in_out() {
     );
 }
 
+#[test]
+fn test_tags() {
+    #[derive(Serialize, Apiv2Schema)]
+    struct Image {
+        data: String,
+        id: u64,
+    }
+
+    #[api_v2_operation(tags = "cats,dogs")]
```

Comment:
I feel like we can add `#[api_v2_tag]` to make our lives easier (and it'll be similar to what we have for `Apiv2Security` derive).
```rust
#[api_v2_tag(name = "cat", description="foo", external_docs="bar")]  // optional name, description and external docs
struct Cat;  // By default, it'll take the struct's exact name
```
... and then we use it in operations like so,
```rust
#[api_v2_operation(tags(Cat, Dog, ...))]
fn my_handler() { /* */ }
```
What do you think? If you have other things to do, we can land this as it is and I'll work on it :smile:

Reply:
Yeah I like this deriving approach as it is more type safe. Though not fully understand how tags would end up in schema. Do you mean that each handler would return list of tags via some additional trait method like operation_tags() and the API would traverse all handlers to compose the full list.

I'm not sure when I will be able to dedicate time for this as we just passed alpha and maintenance is taking some extra time. But as soon as I got this will be on my list for sure.

But still considering above - before merging this in even as intermediate, it would be worth changing macros attributes to the ones which you suggested:
```
#[api_v2_operation(tags(Cat, Dog, ...))]
```
This should be quite reasonable effort which I will try to complete this week. Will just treat them as string for now in this PR but it will make them compatible with the structural tags as you suggested.

Reply:
> Do you mean that each handler would return list of tags via some additional trait method like operation_tags() and the API would traverse all handlers to compose the full list.

That would be a way, yes. We already do this for security definitions and per-handler security stuff.

> This should be quite reasonable effort which I will try to complete this week.

Cool. I'm totally okay with landing this as an intermediate version and if you don't find time, I can implement this at some point soon. Let me know either ways.

Reply:
@dunnock if you don't have time for it I probably have time to make this change this week

Reply:
Thanks @platy ! I'll take it this week, though if you need it sooner pls feel free to PR in my branch.

Reply:
Pushed relevant changes